datanode/src/storage_api.py

```diff
@@ -380,6 +405,149 @@ def _DeleteGridCellMetaData(zoom, x, y, uss_id):
   return result
 
 
+def _GetGridCellsMetaData(zoom, tiles):
+  """Provides an instantaneous snapshot of the metadata for a multiple GridCells
+
+  Args:
+    zoom: zoom level in slippy tile format
+    tiles: array of x,y tiles to retrieve
+  Returns:
+    200 with token and JSON metadata,
+    or the nominal 4xx error codes as necessary.
+  """
+  log.info('Grid cells metadata request instantiated for %sz, %s...',
+           zoom, str(tiles))
+  result = wrapper.get_multi(zoom, tiles)
+  return result
+
+def _PutGridCellsMetaData(zoom, tiles, uss_id):
+  """Updates the metadata stored in multiple GridCells.
+
+    Updates the metadata stored in a multiple GridCell using optimistic locking
+    behavior. Operation fails if the metadata has been updated since
+    GET GridCellsMetadata was originally called (based on sync_token).
+
+  Args:
+    zoom: zoom level in slippy tile format
+    tiles: array of x,y tiles to retrieve
+    uss_id: the plain text identifier for the USS from OAuth
+  Plus posted webargs:
+    sync_token: the composite sync_token retrieved in the
+      original GET GridCellsMetadata,
+    scope: The submitting USS scope for the web service endpoint (used for OAuth
+      access),
+    operation_endpoint: the submitting USS endpoint where all flights in these
+      cells can be retrieved from (variables {zoom}, {x}, and {y} can be used in
+      the endpoint, and will be replaced with the actual grid values),
+    operation_format: The output format for the USS web service endpoint (i.e.
+      NASA, GUTMA),
+    minimum_operation_timestamp: the lower time bound of all of the USSs flights
+      in these grid cells.
+    maximum_operation_timestamp: the upper time bound of all of the USSs flights
+      in these grid cells.
+
+  Returns:
+    200 and a new composite token if updated successfully,
+    409 if there is a locking conflict that could not be resolved, or
+    the other nominal 4xx error codes as necessary.
+  """
+  log.info('Grid cells metadata submit instantiated for %s at %sz, %s...',
+           uss_id, zoom, str(tiles))
+  sync_token = _GetRequestParameter('sync_token', None)
+  if not sync_token and 'sync_token' in request.headers:
```

Comment:
It might be a good idea to move the header check into _GetRequestParameter so we don't have to remember to repeat it whenever we want to get the sync token.

Reply:
Moving it to GetRequestParameter would cause all request parameters to be allowed in the header, and we have only specifically allowed access_token and sync_token.